src/main/kotlin/org/opensearch/replication/action/stop/TransportStopIndexReplicationAction.kt

```diff
@@ -121,7 +121,9 @@ class TransportStopIndexReplicationAction @Inject constructor(transportService:
                 // If paused , we need to make attempt to clear retention leases as Shard Tasks are non-existent
                 if (isPaused) {
                     log.info("Index[${request.indexName}] is in paused state")
-                    attemptRemoveRetentionLease(replMetadata, request.indexName)
+                    val remoteClient = client.getRemoteClusterClient(replMetadata.connectionName)
+                    val retentionLeaseHelper = RemoteClusterRetentionLeaseHelper(clusterService.clusterName.value(), remoteClient)
+                    retentionLeaseHelper.attemptRemoveRetentionLease(clusterService, replMetadata, request.indexName)
                 }
```

Comment:
Since we are removing retention lease removal code from ShardReplicationTask, shouldn't the lease be removed during StopAPI all the time then?

Reply:
Like discussed offline, added retention lease renewal unconditionally and removed the lease cleanup during IndexReplicationTask cleanup.